pkg/apiserver/rest/dualwriter.go

```diff
@@ -80,7 +79,8 @@ type DualWriter interface {
 type DualWriterMode int
 
 const (
-	Mode1 DualWriterMode = iota + 1
+	Mode0 DualWriterMode = iota
```

Comment:
Thanks for the review! I added integration tests for mode 0 to make sure we are taking it into account. 

Mode 0 will only function as a way of representing whether `unifiedStorage` is enabled or not. I think this is fine because it too represents one of the dual writing states, that of not being enabled. To me dual writing doesn't have to mean literally writing to both storages. The advantage of this approach is that it allows us to simplify the signature of `GetAPIGroupInfo`, which was suggested in https://github.com/grafana/grafana/pull/87668#issuecomment-2107333342. This was the best way I found to do so but happy to consider other options of course.